includes/admin/home/notices/class-sensei-home-notices.php

```diff
@@ -0,0 +1,317 @@
+<?php
+/**
+ * File containing Sensei_Home_Notices class.
+ *
+ * @package sensei-lms
+ * @since   $$next-version$$
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class that gathers and produces the local notices for Sensei Home.
+ */
+class Sensei_Home_Notices {
+	const HOME_NOTICE_KEY_PREFIX = 'sensei_home_';
+
+	/**
+	 * The remote data helper.
+	 *
+	 * @var Sensei_Home_Remote_Data_API $remote_data_api
+	 */
+	private $remote_data_api;
+
+	/**
+	 * Cache of local plugin updates.
+	 *
+	 * @var array
+	 */
+	private $local_plugin_updates;
+
+	/**
+	 * Sensei_Home_Notices constructor.
+	 *
+	 * @param Sensei_Home_Remote_Data_API $remote_data_api The remote data helper.
+	 */
+	public function __construct( Sensei_Home_Remote_Data_API $remote_data_api ) {
+		$this->remote_data_api = $remote_data_api;
+	}
+
+	/**
+	 * Add the hooks related to this class.
+	 */
+	public function init() {
+		add_filter( 'sensei_show_admin_notices_' . Sensei_Home::SCREEN_ID, '__return_false' );
+		add_filter( 'sensei_admin_notices', [ $this, 'add_update_notices' ] );
+	}
+
+	/**
+	 * Add the update notices.
+	 *
+	 * @access private
+	 *
+	 * @param array $notices The notices to add the update notices to.
+	 *
+	 * @return array
+	 */
+	public function add_update_notices( $notices ) {
+		if ( ! current_user_can( 'install_plugins' ) ) {
+			return $notices;
+		}
+
+		$data = $this->remote_data_api->fetch();
+		if ( $data instanceof \WP_Error || empty( $data['versions'] ) ) {
```

Comment:
should we also check if `$data['versions']` isset ?

Reply:
Aren't we doing that here?

Reply:
Oh wow! I thought empty dropped warnings! Always learning! 🤓